codestyle/styles.md

```diff
@@ -0,0 +1,633 @@
+  3. [Nesting](#less_nesting)
+
+
+---
+
+- Status of this document: **Draft**
+- Author: Dmitriy Bogolyubov
+- E-mail: dmitriy.bogolyubov@lazada.com
+
+---
+
+Please, feel free to share your own ideas, join discussions and make corrections
+(especially in those parts of the document which have not been filled out yet).
+If you have any objections please do not put them off, it will be much harder to
+add any changes after this document status will be changed from **Draft** to
+**Standard**.
+
+Scrutinize each rule at the point of view your daily work:
+- Will you be able to adhere it in your work?
+- Will it be comfortable for you?
+- Does it work for your team?
+
+If you found a grammar mistake or would like to reformulate
+something for better understanding let me know as well.
+
+---
+
+<a id="base"></a>
+## Base requirements
+
+* UTF-8 encoding without BOM;
+* unix-style line endings;
+* 80-characters line wide where possible;
+* four (4) spaces indents, no tabs;
+* no trailing spaces (empty lines must contain only the line feed symbol);
+* lowercase only, except custom values in quotes, url's and HEX-values;
+* single quotes;
+* one (1) empty line at the end of files;
+
+
+
+<a id="ruleset"></a>
+## Ruleset
+
+The standard ruleset looks like:
+
+```CSS
+[selector] {
+    [property]: [value];
+    [<—declaration—>]
+}
+```
+
+for example:
+
+```CSS
+.block {
+    display: block;
+    background-color: green;
+    color: red;
+}
+```
+
+* each ruleset starts on its own new line;
+* the opening brace (`{`) on the same line as the last selector;
+* one (1) space between the last selector and the opening brace (`{`);
+* each declaration on its own new line indented by four (4) spaces;
+* no empty lines between declarations, except grouping properties;
+* properties and values on the same line;
+* no spaces between property and colon (`:`);
+* one (1) space between colon (`:`) and value;
+* each declaration ends with a semicolon (`;`);
+* no spaces between the value and the semicolon (`;`);
+* the closing brace (`}`) on its own new line, indented as the last
+  selector;
+
+**Grouping selectors**
+
+Two or more selectors have to be separated by comma:
+
+```CSS
+.foo,
+.bar,
+.baz {
+    display: block;
+    color: red;
+}
+```
+
+* each selector on its own new line;
+* no empty lines between grouped selectors;
+* each selector in the list except the last one ends with a comma (`,`);
+* no spaces between selector and comma (`,`);
+
+**Grouping properties**
+
+You can group related properties adding one (1) empty line between groups of
+properties:
+
+```CSS
+.block {
+    position: absolute;
+
+    border-style: solid;
+    border-width: 1px 0 0 1px;
+    border-color: #FFF;
+
+    background-color: #000;
+    background-position: left top;
+}
+```
+
+This record will be wrong because grouped properties are not related to each
+other:
+
+```CSS
+.block {
+    display: block;
+
+    font-size: 30px;
+    background: #EEE;
+}
+```
```

Comment:
The only objection I thought maybe that I don't know an easy way to control compliance with the rule. This is also the reason why I haven't mark this rule as obligatory. Besides that, grouping properties make no sense if we have just, let's say, 3 properties in a ruleset, so maybe you are right. On the other hand this rule helps to make code not so uniform.